plugins/xpressive/expressive_plugin.cpp

```diff
@@ -0,0 +1,884 @@
+		m_parameterA3(1, -1.0f, 1.0f, 0.01f, this, tr("A3")),
+		m_smoothW1(0, 0.0f, 70.0f, 1.0f, this, tr("W1 smoothing")),
+		m_smoothW2(0, 0.0f, 70.0f, 1.0f, this, tr("W2 smoothing")),
+		m_smoothW3(0, 0.0f, 70.0f, 1.0f, this, tr("W3 smoothing")),
+		m_panning1( 1, -1.0f, 1.0f, 0.01f, this, tr("PAN1")),
+		m_panning2(-1, -1.0f, 1.0f, 0.01f, this, tr("PAN2")),
+		m_relTransition(50.0f, 0.0f, 500.0f, 1.0f, this, tr("REL TRANS")),
+		m_W1(GRAPH_LENGTH),m_W2(GRAPH_LENGTH),m_W3(GRAPH_LENGTH),
+		m_exprValid(false, this)
+{
+	m_outputExpression[0]="sinew(integrate(f*(1+0.05sinew(12t))))*(2^(-(1.1+A2)*t)*(0.4+0.1(1+A3)+0.4sinew((2.5+2A1)t))^2)";
+	m_outputExpression[1]="expw(integrate(f*atan(500t)*2/pi))*0.5+0.12";
+}
+
+expressive::~expressive() {
+}
+
+void expressive::saveSettings(QDomDocument & _doc, QDomElement & _this) {
+
+	// Save plugin version
+	_this.setAttribute("version", "0.1");
+	_this.setAttribute("O1", QString(m_outputExpression[0]));
+	_this.setAttribute("O2", QString(m_outputExpression[1]));
+	_this.setAttribute("W1", QString(m_wavesExpression[0]));
+	// Save sample shape base64-encoded
+	QString sampleString;
+	base64::encode( (const char *)m_rawgraphW1.samples(),
+		 m_rawgraphW1.length() * sizeof(float), sampleString );
+	_this.setAttribute( "W1sample", sampleString );
+
+	_this.setAttribute("W2", QString(m_wavesExpression[1]));
+	base64::encode( (const char *)m_rawgraphW2.samples(),
+		 m_rawgraphW2.length() * sizeof(float), sampleString );
+	_this.setAttribute( "W2sample", sampleString );
+	_this.setAttribute("W3", QString(m_wavesExpression[2]));
+	base64::encode( (const char *)m_rawgraphW3.samples(),
+		 m_rawgraphW3.length() * sizeof(float), sampleString );
+	_this.setAttribute( "W3sample", sampleString );
+	m_smoothW1.saveSettings(_doc,_this,"smoothW1");
+	m_smoothW2.saveSettings(_doc,_this,"smoothW2");
+	m_smoothW3.saveSettings(_doc,_this,"smoothW3");
+	m_parameterA1.saveSettings(_doc,_this,"A1");
+	m_parameterA2.saveSettings(_doc,_this,"A2");
+	m_parameterA3.saveSettings(_doc,_this,"A3");
+	m_panning1.saveSettings(_doc,_this,"PAN1");
+	m_panning2.saveSettings(_doc,_this,"PAN2");
+	m_relTransition.saveSettings(_doc,_this,"RELTRANS");
+
+}
+
+void expressive::loadSettings(const QDomElement & _this) {
+
+	m_outputExpression[0]=_this.attribute( "O1").toLatin1();
+	m_outputExpression[1]=_this.attribute( "O2").toLatin1();
+	m_wavesExpression[0]=_this.attribute( "W1").toLatin1();
+	m_wavesExpression[1]=_this.attribute( "W2").toLatin1();
+	m_wavesExpression[2]=_this.attribute( "W3").toLatin1();
+
+	m_smoothW1.loadSettings(_this,"smoothW1");
+	m_smoothW2.loadSettings(_this,"smoothW2");
+	m_smoothW3.loadSettings(_this,"smoothW3");
+	m_parameterA1.loadSettings(_this,"A1");
+	m_parameterA2.loadSettings(_this,"A2");
+	m_parameterA3.loadSettings(_this,"A3");
+	m_panning1.loadSettings(_this,"PAN1");
+	m_panning2.loadSettings(_this,"PAN2");
+	m_relTransition.loadSettings(_this,"RELTRANS");
+
+	int size = 0;
```

Comment:
Not sure for this one, as I don't know the effect of the next function calls, but:

```c++
const int size = 0;
```